agent/cache/cache.go

```diff
@@ -316,37 +316,49 @@ func newGetOptions(tEntry typeEntry, r Request) getOptions {
 // getEntryLocked retrieves a cache entry and checks if it is ready to be
 // returned given the other parameters. It reads from entries and the caller
 // has to issue a read lock if necessary.
-func (c *Cache) getEntryLocked(
-	tEntry typeEntry,
-	key string,
-	info RequestInfo,
-) (entryExists bool, entryValid bool, entry cacheEntry) {
+func (c *Cache) getEntryLocked(tEntry typeEntry, key string, info RequestInfo) (entry cacheEntry) {
```

Comment:
I attempted to make a similar change at some point, but ran into the same test failures you have here. I believe there is a subtle reason that both exists and valid are required.

I ended up taking a bit of a different approach in this branch https://github.com/hashicorp/consul/compare/dnephin/agent-cache-fetch...dnephin/agent-cache-fetch-2 by replacing the function with `isEntryValid()`, which I think improves things a bit, because only one of the call sites cares about the existence of the entry.

Unfortunately that was mixed up with some other changes and never got into a state that was ready for a merge.